lib/common/rand.c

```diff
@@ -0,0 +1,55 @@
+#include <openssl/rand.h>
+#include <inttypes.h>
+#include "h2o/rand.h"
+#include "h2o/string_.h"
+#include "picotls/openssl.h"
+
+static void format_uuid_rfc4122(char *dst, const uint8_t *uuid)
+{
+    // String Representation:
+    // > UUID  = time-low "-" time-mid "-"
+    // >         time-high-and-version "-"
+    // >         clock-seq-and-reserved
+    // >         clock-seq-low "-" node
+    // See also "4.1.2. Layout and Byte Order" for the layout
+    size_t pos = 0;
+#define UUID_ENC_PART(b, p, u, start, last) do { \
```

Comment:
> Considering the fact that this is a closure, maybe use dst, pos, uuid of the enclosing function rather than doing pass-by-name as b, p, u?

✅ done

---

For the other parts, I agree with what you pointed. In fact, `start` and `last` is not matched to the C/C++ convention, and `start - last + 1` to calculate the size seems a bit strange to me as well, but I'd like to keep them for that macro because it is meant to exactly matched to what the spec says in the 4.1.2. "Layout and Byte Order".

Reply:
I see. If that's the concern, I'd suggest using "first" and "last" as the terms. Because "last X" implies that X exists. That's the different from using "end."

Reply:
That's great! Changed to use the terms "first" and "last" 517af66efc43687c4fe14f28cc3b0f95f76ce737